frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx

```diff
@@ -113,25 +110,80 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
 
+  const [dataTemplate, setDataTemplate] = useState<WorkspaceTemplatesQuery>();
+
   const {
-    data: dataTemplate,
     loading: loadingTemplate,
     error: errorTemplate,
-    refetch: refetchTemplate,
+    subscribeToMore: subscribeToMoreTemplates,
   } = useWorkspaceTemplatesQuery({
     variables: { workspaceNamespace },
-    notifyOnNetworkStatusChange: true,
+    onCompleted: setDataTemplate,
   });
 
-  //This useEffect and the reload state are used to simulate the subscription behaviour, it will be removed in the next PR
+  const { instances } = dataInstances?.instanceList ?? {};
+
   useEffect(() => {
-    if (reload && refetchTemplate) {
-      setReload(false);
-      refetchTemplate({ workspaceNamespace });
-    }
-  }, [reload, refetchTemplate, setReload, workspaceNamespace]);
+    if (!loadingTemplate) {
+      subscribeToMoreTemplates({
+        document: updatedWorkspaceTemplates,
+        variables: { workspaceNamespace: `${workspaceNamespace}` },
+        updateQuery: (prev, { subscriptionData }) => {
+          let subscriptionResult: 'modified' | 'added' | 'deleted' | '' = '';
+          const {
+            data,
+          } = subscriptionData as UpdatedWorkspaceTemplatesSubscriptionResult;
+          const newItem = data?.updatedTemplate?.template!;
 
-  const { instances } = dataInstances?.instanceList ?? {};
+          if (!data?.updatedTemplate?.template) return prev;
+
+          if (prev.templateList?.templates) {
+            const oldItem = prev.templateList.templates.find(
+              t =>
+                t?.metadata?.id ===
+                data?.updatedTemplate?.template?.metadata?.id
+            );
+            if (oldItem) {
+              if (JSON.stringify(oldItem) === JSON.stringify(newItem)) {
+                //template have been deleted
+                prev.templateList.templates = prev.templateList?.templates!.filter(
+                  t =>
+                    data?.updatedTemplate?.template?.metadata?.id !==
+                    t?.metadata?.id
+                );
+                subscriptionResult = 'deleted';
+              } else {
+                //template have been modified
+                prev.templateList.templates = prev.templateList?.templates!.map(
+                  t =>
+                    data?.updatedTemplate?.template?.metadata?.id ===
+                    t?.metadata?.id
+                      ? data?.updatedTemplate?.template!
+                      : t!
+                );
+                subscriptionResult = 'modified';
+              }
+            } else {
+              prev.templateList.templates = [
+                ...prev.templateList.templates,
+                data?.updatedTemplate?.template!,
+              ];
+              subscriptionResult = 'added';
+            }
+          }
+
+          notification['info']({
+            message: newItem.spec?.name,
+            description: `This Template have been ${subscriptionResult}`,
```

Comment:
```suggestion
            message: 'Workspace update!',
            description: `Template <i>${newItem.spec?.name}</i> has been ${subscriptionResult}`,
```
I'd suggest something like this. Please check whether html tags are supported in the notification, if not just remove them.

Reply:
Don't know if @giorio94 has any better suggestion

Reply:
Not really. Could just specify the workspace in the message, if that makes sense.